src/mds/Server.cc

```diff
@@ -1456,6 +1456,8 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
     req->get_connection()->send_message(reply);
   }
 
+  assert (reply);
```

Comment:
This wouldn't be right as the prior if can change `reply` to `NULL`. You need to move this assert to the `if` below.